medical/models/medical_patient.py

```diff
@@ -95,6 +95,18 @@ def action_invalidate(self):
             ('z', 'law marriage'),
         ])
 
+    @api.multi
+    @api.onchange('state_id')
+    def onchange_state(self):
+        """ Hack to allow for onchange_state on partner.
+        Uses current user's partner, as there should be no correlation
+        between the partner and the response from this method anyways
+        """
+        res = self.env.user.partner_id.onchange_state(self.state_id.id)
```

Comment:
Old onchange methods are an exception to @api.multi specification (it was bad designed). They pass ids argument (or recordset in new API), but they can't use it, because all the changes are on the fly on even non created records, so it's safe to use an empty recordset. If you find that one of these methods is using the recordset, then the method is incorrect by definition.